fem/field_interpolant.cpp

```diff
@@ -0,0 +1,441 @@
+// Copyright (c) 2010-2020, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+// Implementation of Field Interpolants and necessary (Vector)QuadratorIntegrators
+
+#include "field_interpolant.hpp"
+#include "../linalg/densemat.hpp"
+#include "fem.hpp"
+
+namespace mfem
+{
+
+void VectorQuadratureIntegrator::AssembleRHSElementVect(const FiniteElement &fe,
+                                                        ElementTransformation &Tr,
+                                                        Vector &elvect)
+{
+   const int nqp = IntRule->GetNPoints();
+   const int vdim = vqfc.GetVDim();
+   const int ndofs = fe.GetDof();
+   Vector shape(ndofs);
+   Vector temp(vdim);
+   elvect.SetSize(vdim * ndofs);
+   elvect = 0.0;
+   for (int q = 0; q < nqp; q++)
+   {
+      const IntegrationPoint &ip = IntRule->IntPoint(q);
+      Tr.SetIntPoint(&ip);
+      const double w = Tr.Weight() * ip.weight;
+      vqfc.Eval(temp, Tr, ip);
+      fe.CalcShape(ip, shape);
+      for (int ind = 0; ind < vdim; ind++)
+      {
+         for (int nd = 0; nd < ndofs; nd++)
+         {
+            elvect(nd + ind * ndofs) += w * shape(nd) * temp(ind);
+         }
+      }
+   }
+}
+
+void QuadratureIntegrator::AssembleRHSElementVect(const FiniteElement &fe,
+                                                  ElementTransformation &Tr,
+                                                  Vector &elvect)
+{
+   const int nqp = IntRule->GetNPoints();
```

Comment:
This won't work if `IntRule` is null, which happens with the default constructor.

Reply:
```IntRule``` is now checked for to make sure it's not null. If it is it does the typical construction of one that you see in other parts of the code. Also, the ```IntRule``` within these LFIntegrators are taken directly from the ```QuadratureFunction``` rather than using the class one just in case the QuadratureSpace has variable IntegrationRules in each element.